catalog/spatial/wfs/1.1.0/spatial-wfs-v1_1_0-source/src/main/java/org/codice/ddf/spatial/ogc/wfs/v110/catalog/source/WfsFilterDelegate.java

```diff
@@ -529,20 +531,43 @@ private FilterType buildPropertyIsFilterTypeForProperty(
       Object literal,
       PROPERTY_IS_OPS propertyIsType,
       boolean isCaseSensitive) {
-    FilterType returnFilter = new FilterType();
-    FeatureAttributeDescriptor attrDesc =
-        (FeatureAttributeDescriptor) featureMetacardType.getAttributeDescriptor(propertyName);
-    if (attrDesc.isIndexed()) {
-      returnFilter.setComparisonOps(
-          createPropertyIsFilter(
-              attrDesc.getPropertyName(), literal, propertyIsType, isCaseSensitive));
-      return returnFilter;
+    final String featurePropertyName =
+        mapQueryPropertyToFeatureProperty(propertyName, this::isWfsFeatureProperty);
+    if (featurePropertyName != null) {
+      final FilterType filter = new FilterType();
+      filter.setComparisonOps(
+          createPropertyIsFilter(featurePropertyName, literal, propertyIsType, isCaseSensitive));
+      return filter;
+    }
+
+    return null;
+  }
+
+  private String mapQueryPropertyToFeatureProperty(
+      final String queryProperty, final Predicate<String> isWfsFeatureProperty) {
+    final String featurePropertyName =
+        Optional.ofNullable(metacardMapper.getFeatureProperty(queryProperty)).orElse(queryProperty);
+
+    if (!isWfsFeatureProperty.test(featurePropertyName)) {
+      LOGGER.debug(
+          "The property '{}' could not be mapped to a feature property of feature type '{}'.",
+          featurePropertyName,
+          featureMetacardType.getFeatureType());
+      return null;
+    }
+
+    final FeatureAttributeDescriptor featureAttributeDescriptor =
+        (FeatureAttributeDescriptor)
+            featureMetacardType.getAttributeDescriptor(featurePropertyName);
+    if (!featureAttributeDescriptor.isIndexed()) {
+      throw new IllegalArgumentException(
+          String.format(PROPERTY_NOT_QUERYABLE, featurePropertyName));
```

Comment:
We throw PROPERTY_NOT_QUERYABLE when it's not indexed, but just return `null` if we can't map it. Should we also throw an exception if the property isn't mapped? Or alternatively, should we just return `null` if it's not indexed.

Reply:
If the property is not indexed then that means it is configured as a non-queryable property in the WFS source configuration. The metatype description for `nonQueryableProperties` states an exception will be thrown if one of the properties is found in the query so we shouldn't change that behavior.

Returning null means that branch of the filter will be dropped. Note that I did not change the filter delegate's behavior with respect to returning null. It already returns null if the property does not exist on the WFS feature type. That being said, I wonder if that is the right thing to do because that could result in the WFS source returning results that don't match the original query (e.g. if one clause of an AND is dropped because the WFS source doesn't have that property). It may be more appropriate to throw an exception instead. Thoughts?

Reply:
I've updated it to throw an exception instead.